### PR TITLE
Better compile error output when using arguments instead of types 

### PR DESCRIPTION
**Pull request by jean-lourenco**
Following @estebank sugestion on issue https://github.com/rust-lang/rust/issues/18945#issuecomment-331251436

### CODE REVIEW

File: src/test/ui/did_you_mean/issue-40396.stderr [hunk 1]
```diff
@@ -4,31 +4,31 @@ error: chained comparison operators require parentheses
 12 |     println!("{:?}", (0..13).collect<Vec<i32>>());
    |                                     ^^^^^^^^
    |
-   = help: use `::<...>` instead of `<...>` if you meant to specify type arguments
+   = help: use `::<...>` instead of `<...>` if you meant to specify type arguments or use `(...)` if you meant to specify fn arguments
```
Comment (petrochenkov) [inline src/test/ui/did_you_mean/issue-40396.stderr]: This will probably look better as a second `help:` on a separate line.

Comment (nikomatsakis) [reply src/test/ui/did_you_mean/issue-40396.stderr]: I think I also would prefer a second help

**Comment by jean-lourenco** [comment]
The recommended changes were implemented.

**Comment by estebank** [comment]
@jean-lourenco the broken test is to be expected and correct, if you could fix it and squash all your commits into one we can approve.

**Comment by nikomatsakis** [comment]
Looks like you have to update `parse-fail/require-parens-for-chained-comparison.rs` to something like this:

```rust
    f<X>();
    //~^ ERROR: chained comparison operators require parentheses
    //~| HELP: use `::<...>` instead of `<...>`
    //~| HELP: or use `(...)`
```

**Comment by jean-lourenco** [comment]
Now the tests are green 🎉
Thank you guys for your help!

**Comment by nikomatsakis** [comment]
Thanks @jean-lourenco

**Pull request merged**